phoenix-core/src/main/java/org/apache/phoenix/coprocessor/metrics/MetricsPhoenixCoprocessorSourceFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.apache.phoenix.coprocessor.metrics;
+
+public class MetricsPhoenixCoprocessorSourceFactory {
+
+    private static final MetricsPhoenixCoprocessorSourceFactory
+            INSTANCE = new MetricsPhoenixCoprocessorSourceFactory();
+    private MetricsPhoenixTTLSource phoenixTTLSource;
+
+    public static MetricsPhoenixCoprocessorSourceFactory getInstance() {
+        return INSTANCE;
+    }
+
+    public synchronized MetricsPhoenixTTLSource getPhoenixTTLSource() {
```

Comment:
Do double-check locking instead, to improve performance and reduce contention here.

Reply:
@jpisaac can we address this please?